### ISSUE TITLE
No profile auto switch with Gnome Shell 46.5

### ISSUE THREAD

**Issue by szpak** [yurenchen000/terminator-autotheme-plugin#1]
Thanks for the plugin. It makes it easier to switch a profile globally for all the terminals. However, the ability to auto switch a profile when a dark mode is enabled doesn't work with Gnome Shell 46.5 on Fedora 40.

When switched to the dark mode some other apps, such as Gnome Terminal or Firefox change the theme, but in "Auto Theme Setting", the "Light" is in the green rectangle, even though "Mode" is switched to "Auto". It even still detects "Light theme" when Terminator is restarted.

Manually changing the mode to "Dark" works as expected. Terminator 2.1.4.

Do you know what can be wrong?

**Comment by yurenchen000** [comment]
which plugin version do you use?

some known issue:
  - plugin may not auto load right status after terminator restart (need open settings manually, or change system theme to trigger change)

i didn't try it on `Gnome Shell 46.5 on Fedora 40` yet
 - i guess the dark theme detect maybe not work well

<br>

can you turn on the debug log and post here?
- just comment out the `print = lambda *a:None` at line 29
- and restart terminator in other terminal via `terminaotr -u` to get the running log

**Comment by yurenchen000** [comment]
I'm curious about the log print on your system
> print('== on_theme_name change:', theme_name, theme_variant)

<br>

And maybe you can try `1.2.0` dev version

**Comment by szpak** [comment]
> which plugin version do you use?

The latest from master the then. I've missed the "dev" branch.

With 1.2.0 it works, thanks!

(although Vte and Gtk settings seem to control slightly different aspect of Terminator)


The logs you requested:

```
# after "terminator -u"
--- load_config: {'light': 'default', 'dark': 'dark', 'mode': 'Auto', 'variant': 'auto'}
=== load_config: default dark Auto auto
++ setup_theme_monit

# after opening the plugin configuration
--theme_name: Adwaita False
load mode: False
--change_theme: False default
-- teardown_theme_monit
++ setup_theme_monit
profiles: ['default', 'dark']
--- load_config: {'light': 'default', 'dark': 'dark', 'mode': 'Auto', 'variant': 'auto'}
=== load_config: default dark Auto auto
--theme_name: Adwaita False
-- Cancel/close clicked.

# after enabling the Gnome dark theme
--_on_theme_name_changed: <Gtk.Settings object at 0x7fa42289fd00 (GtkSettings at 0x55eb164d5be0)> <GParamString 'gtk-theme-name'>
== on_theme_name change: Adwaita True
--change_theme: True dark

# on switching Ghome theme to white/light
--_on_theme_name_changed: <Gtk.Settings object at 0x7fa42289fd00 (GtkSettings at 0x55eb164d5be0)> <GParamString 'gtk-theme-name'>
== on_theme_name change: Adwaita False
--change_theme: False default
```

Would you like me to check anything else (to allow you to tweak up the plugin)?

**Comment by yurenchen000** [comment]
> although Vte and Gtk settings seem to control slightly different aspect of Terminator

Yes, I want them independently control `terminal color` and `Gtk UI color` ,
But they influence each other

(i haven't find a good way yet , maybe i should give up that mind and merge the 2 options into 1.

<br>

> Would you like me to check anything else

Thanks, nothing else for now.
it works same as on my environment now (ubuntu 22 lts).

**Comment by szpak** [comment]
> (i haven't find a good way yet , maybe i should give up that mind and merge the 2 options into 1.

It's rather not a problem to switch both of them to a requested value. Maybe with some other "window managers" that separation would be useful.

**Comment by yurenchen000** [comment]
> independently control terminal color and Gtk UI color ,
But they influence each other

this has improved in [release 1.2.2](../releases/tag/1.2.2)
